docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,966 @@
+   * Check for critical process – 5 mins timeout – check every 20 secs
+   * Check for transceiver status –– 5 mins timeout – check every 20 seconds
+   * Check for pmon status
+   * Check for reboot cause 
+   * Reboot is successful
+   
+Reboot Test Case for DPU:
+ * After the exisiting case, Power on all the DPUs using `config chassis modules startup <DPU_Number>`
+ * Wait for DPUs to be up
+ * Use command `show chassis modules status` to get DPU status
+ * Get the number of DPU modules from PMON APIs - get_num_modules()
+   
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+
+root@sonic:/home/cisco# reboot
+root@sonic:/home/cisco#
+root@sonic:/home/cisco# config chassis modules startup <DPU_Number>
+root@sonic:/home/cisco#
+root@sonic:/home/cisco#
+root@sonic:/home/cisco# show chassis modules status
+  Name    Description    Physical-Slot    Oper-Status    Admin-Status    Serial
+------  -------------  ---------------  -------------  --------------  --------
+  DPU0            N/A               -1         Online              up       N/A
+  DPU1            N/A               -1         Online              up       N/A
+  DPU2            N/A               -1         Online              up       N/A
+  DPU3            N/A               -1         Online              up       N/A
+  DPU4            N/A               -1         Online              up       N/A
+  DPU5            N/A               -1         Online              up       N/A
+  DPU6            N/A               -1         Online              up       N/A
+  DPU7            N/A               -1         Online              up       N/A
+```
+
+#### Pass/Fail Criteria 
+ *  Verify number of DPUs from api and number of DPUs shown in the cli output.
+
+
+## Objectives of API Test Cases
+
+|    | **Test Case**   | **Intention**                              | **Comments** |
+| ---------- | ---------- | ---------------------------------------- | ---------- |
+| 1.1 | Check SmartSwitch specific ChassisClass APIs      | To verify the newly implemented SmartSwitch specific ChassisClass APIs | |
+| 1.2 | Check modified ChassisClass APIs for SmartSwitch       |  To verify the existing ChassisClass APIs that undergo minor changes with the addition of SmartSwitch| |
+| 1.3 | Check DpuModule APIs for SmartSwitch       |  To verify the newly implemented  DpuModule APIs for SmartSwitch| |
+| 1.4 | Check modified ModuleClass APIs for SmartSwitch       |  To verify the existing ModuleClass APIs that undergo minor changes with the addition of SmartSwitch| 
+
+## API Test Cases 
+
+### 1.1 Check SmartSwitch specific ChassisClass APIs
+
+#### Steps
+ * Execute the following APIs on SmartSwitch
+ * get_dpu_id(self, name):
+    * Provide name (Example: DPU0 - Get it from platform.json file)
+    * This API should return an integer from 1-8 (check it against platform.json)
+ * is_smartswitch(self):
+    * This API should return True
+ * get_module_dpu_data_port(self, index):
```

Comment:
how can we check if this output is correct/expected? from platform.json?

Reply:
Yes comparing against platform.json